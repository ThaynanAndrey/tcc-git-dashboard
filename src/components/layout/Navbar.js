import React from 'react';
import { Link } from 'react-router-dom';

import SignedInLinks from './SignedInLinks';
import SignedOutLinks from './SignedOutLinks';
import { isAuthenticated, logout } from '../../store/actions/authAction';

const styles = {
  marginRight: "50px",
  marginLeft: "50px"
};

/**
 * Stateless Component with the function of presenting a
 * main menu for redirects within the application.
 * 
 * @author Thaynan Nunes
 */
const Navbar = () => {
  
  /**
   * Sign out user and redirect to login page.
   */
  const signOut = () => {
    logout();
  };

  const links = isAuthenticated() ? <SignedInLinks signOut={signOut}/> : <SignedOutLinks />;
  
  return (
    <div className="navbar-fixed">
      <nav className="nav-wrapper blue-grey darken-2">
        <div style={styles}>
          <Link to='/' className="brand-logo">Git Dashboard</Link>
          { links }
        </div>
      </nav>
    </div>
  )
}

export default Navbar;