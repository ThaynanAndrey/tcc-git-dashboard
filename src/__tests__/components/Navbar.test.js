import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import Navbar from '../../components/layout/Navbar';

describe('Test Navbar', () => {
  it('renders correctly a Navbar', () => {
    const tree = shallow(
      <Navbar user={{}} logout={jest.fn()} />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });
})