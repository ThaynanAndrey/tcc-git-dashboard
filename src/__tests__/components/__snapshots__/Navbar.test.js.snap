// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navbar renders correctly a Navbar 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
